cpu/atmega_common/periph/gpio.c

```diff
@@ -237,61 +293,114 @@ void gpio_write(gpio_t pin, int value)
     }
 }
 
+static inline void pcint_handler(uint8_t port_num, uint8_t pin_num)
+{
+    int gpio_state = gpio_read( GPIO_PIN(port_num, pin_num) );
+    gpio_flank_t flank = pcint_flank[ port_num * 8 + pin_num ];
+
+    if (flank == GPIO_BOTH || (gpio_state && flank == GPIO_RISING) || (!gpio_state && flank == GPIO_FALLING)) {
+        __enter_isr();
+        pcint[port_num * 8 + pin_num].cb( pcint[port_num * 8 + pin_num].arg );
+        __exit_isr();
+    }
+}
+
+static inline uint8_t pcint_port_pin(volatile uint8_t *reg)
```

Comment:
This doesn't work, imagine PC3 and PC0 are 1, there's an pcint enabled for PC1. You'll call `pcint_handler` with C and 2.

Reply:
Can't follow on this one

Reply:
This should be eliminated once we keep track of the state anyway!